### PR TITLE
github-electron: Update to v1.2.7

### PR DESCRIPTION
**Pull request by rhysd**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://github.com/electron/electron/releases .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by miniak** [comment]
@rhysd, can you please also fix the "Wether" typos to "Whether"? It's in multiple places.

**Comment by miniak** [comment]
@rhysd and also this one:

```
beginFrameSubscription(onlyDirty: boolean, callback: FrameSubscriptionCallback): void;
beginFrameSubscription(callback: FrameSubscriptionCallback): void;

type FrameSubscriptionCallback = (
	/**
	 * The frameBuffer is a Buffer that contains raw pixel data.
	 * On most machines, the pixel data is effectively stored in 32bit BGRA format,
	 * but the actual representation depends on the endianness of the processor
	 * (most modern processors are little-endian, on machines with big-endian
	 * processors the data is in 32bit ARGB format).
	 */
	frameBuffer: Buffer,
	/**
	 * Describes which part of the page was repainted. If onlyDirty is set to true,
	 * frameBuffer will only contain the repainted area. onlyDirty defaults to false.
	 */
	dirtyRect: Bounds
) => void;
```

File: github-electron/github-electron.d.ts [hunk 1]
```diff
@@ -3763,6 +3796,11 @@ declare namespace Electron {
 		 */
 		showDefinitionForSelection(): void;
 		/**
+		 * Sets the item as dragging item for current drag-drop operation, file is the absolute path of
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: I would move the file/icon documentation to the DragItem type.

File: github-electron/github-electron.d.ts [hunk 2]
```diff
@@ -341,6 +341,10 @@ declare namespace Electron {
 		 * This method can only be called before app is ready.
 		 */
 		disableHardwareAcceleration(): void;
+		/**
+		 * @returns whether current desktop environment is Unity launcher. (Linux)
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: In the other places, we use "Note: This API is only available on Linux."
Or maybe we could just change it to this shorter version everywhere instead.
The problem is also that some methods only exist on certain platforms, others exist, but don't have any effect.

**Comment by miniak** [comment]
@rhysd <webview>.capturePage is missing

**Comment by miniak** [comment]
@rhysd app.isAccessibilitySupportEnabled is missing (https://github.com/electron/electron/blob/master/docs/api/app.md#appisaccessibilitysupportenabled-macos-windows)

File: github-electron/github-electron.d.ts [hunk 3]
```diff
@@ -415,6 +419,16 @@ declare namespace Electron {
...
 		getBadge(): string;
 		/**
+		 * Sets the counter badge for current app. Setting the count to 0 will hide the badge.
+		 *
+		 * @returns True when the call succeeded, otherwise returns false.
+		 */
+		setBadgeCount(count: number): boolean;
+		/**
+		 * @returns The current value displayed in the counter badge.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 4]
```diff
@@ -415,6 +419,16 @@ declare namespace Electron {
 		 */
 		getBadge(): string;
 		/**
+		 * Sets the counter badge for current app. Setting the count to 0 will hide the badge.
+		 *
+		 * @returns True when the call succeeded, otherwise returns false.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 5]
```diff
@@ -1541,10 +1555,23 @@ declare namespace Electron {
 			html?: string;
 			image?: NativeImage;
 		}, type?: ClipboardType): void;
+		/**
+		 * @returns An Object containing title and url keys representing the bookmark in the clipboard.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 6]
```diff
@@ -1541,10 +1555,23 @@ declare namespace Electron {
...
+		readBookmark(): Bookmark;
+		/**
+		 * Writes the title and url into the clipboard as a bookmark.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 7]
```diff
@@ -160,6 +160,10 @@ declare namespace Electron {
 		 * Emitted when the gpu process crashes.
 		 */
 		on(event: 'gpu-process-crashed', listener: Function): this;
+		/**
+		 * Emitted when Chrome's accessibility support changes.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 8]
```diff
@@ -345,6 +349,14 @@ declare namespace Electron {
 		 * @returns whether current desktop environment is Unity launcher. (Linux)
 		 */
 		isUnityRunning(): boolean;
+		/**
+		 * @returns an Object with the login item settings of the app.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

File: github-electron/github-electron.d.ts [hunk 9]
```diff
@@ -345,6 +349,14 @@ declare namespace Electron {
...
+		getLoginItemSettings(): LoginItemSettings;
+		/**
+		 * Set the app's login item settings.
+		 */
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: Missing platforms note

**Comment by miniak** [comment]
@rhysd Sorry for taking so long to review your changes, I was quite busy.

**Comment by rhysd** [comment]
@miniak Thank you for the detailed review.  I'll fix them today!

**Comment by rhysd** [comment]
@miniak I fixed all points you pointed out :smiley:

**Comment by rhysd** [comment]
Although CI status is not being added, I confirmed tests passed with `tsc` v2.0.0.

**Comment by rhysd** [comment]
@miniak Currently not planning.  I'll be happy if you help.

File: github-electron/github-electron.d.ts [hunk 10]
```diff
@@ -4643,6 +4662,11 @@ declare namespace Electron {
 		 */
 		getWebContents(): WebContents;
 		/**
+		 * Captures a snapshot of the webview's page. Same as webContents.capturePage([rect, ]callback).
+		 */
+		capturePage(callback: (image: NativeImage) => void): void;
+		capturePage(rect: Rectangle, callback: (image: NativeImage) => void): void;
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: I would include the same description here, it's done this way in other places to keep code completion with docs working consistently for all method overrides.

File: github-electron/github-electron.d.ts [hunk 11]
```diff
@@ -350,6 +350,15 @@ declare namespace Electron {
 		 */
 		isUnityRunning(): boolean;
 		/**
+		 * Returns a Boolean, true if Chrome's accessibility support is enabled, false otherwise.
```
Comment (miniak) [inline github-electron/github-electron.d.ts]: I would change it to "@returns If Chrome's accessibility support is enabled". Specifying the return type is not necessary, that's visible from the method signature.

**Comment by miniak** [comment]
@rhysd There is one we wether -> whether typo remaining

```
-		 * These flags indicate wether the renderer believes it is able to perform the corresponding action.
+		 * These flags indicate whether the renderer believes it is able to perform the corresponding action.
```

**Comment by miniak** [comment]
@rhysd Please also fix the https://github.com/electron/electron/blob/master/docs/api/web-contents.md#contentsbeginframesubscriptiononlydirty-callback as suggested before. The `onlyDirty` argument of `beginFrameSubscription` is optional, therefore we should introduce the `FrameSubscriptionCallback` type to be able to use it in both method overloads.

File: github-electron/github-electron.d.ts (hunk 10 above)
Comment (rhysd) [inline github-electron/github-electron.d.ts]: OK, I'll add the same description

**Comment by rhysd** [comment]
Ah, I forgot the case where specifying `onlyDirty == false` explicitly.

**Comment by rhysd** [comment]
@miniak I fixed!

**Comment by miniak** [comment]
(+1)

**Comment by miniak** [comment]
@mhegazy All the issue have been resolved.

**Comment by mhegazy** [comment]
thanks for the review.

**Pull request merged**